### PR TITLE
Merge 5.1.1minus-19

### PR DESCRIPTION
**Pull request by liam923**
Merge flambda changes for 5.1.1minus-19 into Merlin. I recommend skipping the first three commits in review, as they run the import script, perform automatic merges, and then check in remaining conflicts. The following commit then resolves conflicts.

### CODE REVIEW

File: src/analysis/construct.ml [hunk 1]
```diff
@@ -106,7 +106,7 @@ module Util = struct
   (** [find_values_for_type env typ] searches the environment [env] for
   {i values} with a return type compatible with [typ] *)
   let find_values_for_type env typ =
-    let aux name path value_description acc =
+    let aux name path value_description _modalities acc =
```
Comment (liam923) [inline src/analysis/construct.ml:109]: Should `_modalities` be used to filter values with the wrong modalities?

File: src/analysis/polarity_search.ml [hunk 2]
```diff
@@ -122,7 +122,7 @@ let directories ~global_modules env =
 
 let execute_query query env dirs =
   let direct dir acc =
-    Env.fold_values (fun _ path desc acc ->
+    Env.fold_values (fun _ path desc _ acc ->
```
Comment (liam923) [inline src/analysis/polarity_search.ml:125]: I'm not sure if it's appropriate to ignore modalities here - I don't know what this file does.

File: src/analysis/ptyp_of_type.ml [hunk 3]
```diff
@@ -132,17 +132,28 @@ and extension_constructor id {
...
+    let modality_str = Format.asprintf "%a" Mode.Modality.print modality in
+    Location.mknoloc @@ Parsetree.Modality modality_str)
+and value_description id { val_type; val_kind=_; val_loc; val_attributes; val_modalities; _ } =
   let type_ = core_type val_type in
   {
     Parsetree.pval_name = var_of_id id;
     pval_type = type_;
     pval_prim = [];
     pval_attributes = val_attributes;
+    pval_modalities = const_modalities @@ Mode.Modality.Value.to_const_exn val_modalities;
```
Comment (liam923) [inline src/analysis/ptyp_of_type.ml:148]: Is it safe to call `to_const_exn` here? It seems like there should be a better way.

File: src/analysis/ptyp_of_type.ml [hunk 4]
```diff
@@ -132,17 +132,28 @@ and extension_constructor id {
     ~args:(constructor_arguments ext_args)
     ?res:(Option.map ~f:core_type ext_ret_type)
     (var_of_id id)
-and value_description id { val_type; val_kind=_; val_loc; val_attributes; _ } =
+and const_modalities modalities =
+  modalities
+  |> Mode.Modality.Value.Const.to_list
+  |> List.map ~f:(fun modality ->
+    let modality_str = Format.asprintf "%a" Mode.Modality.print modality in
```
Comment (liam923) [inline src/analysis/ptyp_of_type.ml:139]: Will `Mode.Modality.print` give syntactically legal modalities?

File: tests/jq [hunk 5]
```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env bash
```
Comment (liam923) [inline tests/jq:1]: This file is janky, but it was my best solution to get tests running. I'm happy to have a conversation about it if this comment is insufficient.

File: tests/jq [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+      fi
+
+      # perform logic to track whether we're in a string
+      if [[ "$in_string" == true ]]; then
+        if [[ "$last_char" != '\' && "$char" == '"' ]]; then
+          in_string=false
+          last_char='"'
+        elif [[ "$last_char" == '\' && "$char" == '\' ]]; then
+          in_string=false
+          last_char='real_\'
```
Comment (ncik-roberts) [inline tests/jq:47]: Why is `real` here?

**Review by ncik-roberts — COMMENTED** [review]
A few minor comments.

**Review by ncik-roberts — APPROVED** [review]
Suppressing the changes in test output looks right to me, thanks.

**Comment by ncik-roberts** [comment]
(We discussed, and Liam is going to extract the hacky script &mdash; which we've written before &mdash; into a function to be called at the sites in tests that want to call jq on the output of the merlin test script.)

File: src/ocaml/utils/config.ml [hunk 7]
```diff
@@ -32,28 +32,28 @@ let version = Sys.ocaml_version
...
   else
-    "Caml2021Z532"
-and ast_impl_magic_number = "Caml1999M033"
-and ast_intf_magic_number = "Caml1999N033"
-and cmxs_magic_number = "Caml1999D532"
-and cmt_magic_number = "Caml1999T532"
-and cms_magic_number = "Caml1999S532"
+    "Caml2021Z533"
+and ast_impl_magic_number = "Caml1999M533"
+and ast_intf_magic_number = "Caml1999N533"
```
[the remark this answers is not in the document]
Comment (liam923) [reply src/ocaml/utils/config.ml:53]: The compiler makes the same change between minus-18 and minus 19

File: tests/jq (hunk 6 above)
Comment (liam923) [reply tests/jq]: outdated

File: tests/jq [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+# This script is a wrapper around the jq command line utility. It replaces \n occurrences
+# inside string literals with actual newlines for readability. This results in invalid
+# json. In order for jq to properly parse such output (to allow jq commands to be
+# chained), it replaces newlines within string with \n when reading from stdin. jq does
+# not handle 
+
+# This script is on the PATH and named jq, but we want to get the jq that would normally
+# be on the path. We remove this one from the path
+DIR=$(dirname "$0")
+export PATH=$(echo "$PATH" | sed "s#$DIR:##g")
```
[the remark this answers is not in the document]
Comment (liam923) [reply tests/jq]: outdated

File: tests/jq [hunk 9]
```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env bash
+
+# This script is a wrapper around the jq command line utility. It replaces \n occurrences
+# inside string literals with actual newlines for readability. This results in invalid
+# json. In order for jq to properly parse such output (to allow jq commands to be
+# chained), it replaces newlines within string with \n when reading from stdin. jq does
+# not handle
```
[the remark this answers is not in the document]
Comment (liam923) [reply tests/jq]: outdated

File: src/kernel/mconfig.ml [hunk 10]
```diff
@@ -439,7 +442,7 @@ let ocaml_ignored_flags = [
   "-dflambda-no-invariants"; "-dflambda-verbose"; "-dinstr"; "-dinterf";
   "-dlambda"; "-dlinear"; "-dlive"; "-dparsetree"; "-dprefer"; "-dshape";
   "-drawclambda"; "-drawflambda"; "-drawlambda"; "-dreload"; "-dscheduling";
-  "-dsel"; "-dsource"; "-dspill"; "-dsplit"; "-dstartup"; "-dtimings";
+  "-dsel"; "-dsource"; "-dspill"; "-dsplit"; "-dstartup"; "-dtimings"; "-dprofile";
```
[the remark this answers is not in the document]
Comment (liam923) [reply src/kernel/mconfig.ml:445]: Yep I think this was mistakenly forgotten at some point in the past. I noticed it while adding another flag.

File: tests/jq (hunk 5 above)
Comment (liam923) [reply tests/jq]: This has been removed

**Pull request merged**